cni/network/network.go

```diff
@@ -553,10 +580,20 @@ func (plugin *netPlugin) Delete(args *cniSkel.CmdArgs) error {
 		return err
 	}
 
-	// Call into IPAM plugin to release the endpoint's addresses.
-	nwCfg.Ipam.Subnet = nwInfo.Subnets[0].Prefix.String()
-	for _, address := range epInfo.IPAddresses {
-		nwCfg.Ipam.Address = address.IP.String()
+	if !nwCfg.MultiTenancy {
+		// Call into IPAM plugin to release the endpoint's addresses.
+		nwCfg.Ipam.Subnet = nwInfo.Subnets[0].Prefix.String()
+		for _, address := range epInfo.IPAddresses {
+			nwCfg.Ipam.Address = address.IP.String()
+			err = plugin.DelegateDel(nwCfg.Ipam.Type, nwCfg)
+			if err != nil {
+				err = plugin.Errorf("Failed to release address: %v", err)
+				return err
+			}
+		}
+	} else if epInfo.EnableInfraVnet {
+		nwCfg.Ipam.Subnet = nwInfo.Subnets[0].Prefix.String()
```

Comment:
Get the Subnet using information from CNS. That will be same in windows and linux.